From 2d74d34d316d150a0717fc369ecff09dab18545f Mon Sep 17 00:00:00 2001
From: Jouni Malinen <j@w1.fi>
Date: Fri, 27 Dec 2024 23:51:55 +0200
Subject: [PATCH] Clear the pending EAPOL RX on disconnection

Avoid any potentially unexpected behavior if the postponed EAPOL RX
could end up being processed in a different association that the one in
which the frame was postponed.

Signed-off-by: Jouni Malinen <j@w1.fi>
---
 wpa_supplicant/events.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/wpa_supplicant/events.c b/wpa_supplicant/events.c
index 77bc3c7a9..bab214572 100644
--- a/wpa_supplicant/events.c
+++ b/wpa_supplicant/events.c
@@ -436,6 +436,9 @@ void wpa_supplicant_mark_disassoc(struct wpa_supplicant *wpa_s)
 
 	wpa_s->ssid_verified = false;
 	wpa_s->bigtk_set = false;
+
+	wpabuf_free(wpa_s->pending_eapol_rx);
+	wpa_s->pending_eapol_rx = NULL;
 }
 
 
-- 
2.43.0

